src/test/javascript/spec/component/course/course-group.component.spec.ts

```diff
@@ -75,130 +68,145 @@ describe('Course Management Detail Component', () => {
-        comp = fixture.componentInstance;
-        courseService = fixture.debugElement.injector.get(CourseManagementService);
-        userService = fixture.debugElement.injector.get(UserService);
+        })
+            .compileComponents()
+            .then(() => {
+                fixture = TestBed.createComponent(CourseGroupComponent);
+                comp = fixture.componentInstance;
+                courseService = TestBed.inject(CourseManagementService);
+                userService = TestBed.inject(UserService);
+            });
     });
 
-    afterEach(function () {
-        sinon.restore();
+    afterEach(() => {
+        jest.restoreAllMocks();
     });
 
     it('should initialize', () => {
         fixture.detectChanges();
-        expect(CourseGroupComponent).to.be.ok;
+        expect(CourseGroupComponent).toBeTruthy();
     });
 
     describe('OnInit', () => {
         it('should load all course group users', () => {
-            const getUsersStub = sinon.stub(courseService, 'getAllUsersInCourseGroup');
-            getUsersStub.returns(of(new HttpResponse({ body: [courseGroupUser] })));
+            const getUsersStub = jest.spyOn(courseService, 'getAllUsersInCourseGroup').mockImplementation();
+            getUsersStub.mockReturnValue(of(new HttpResponse({ body: [courseGroupUser] })));
             fixture.detectChanges();
             comp.ngOnInit();
-            expect(comp.course).to.deep.equal(course);
-            expect(comp.courseGroup).to.deep.equal(courseGroup);
-            expect(getUsersStub).to.have.been.called;
+            expect(comp.course).toEqual(course);
+            expect(comp.courseGroup).toEqual(courseGroup);
+            expect(getUsersStub).toHaveBeenCalled();
         });
     });
 
     describe('searchAllUsers', () => {
         let loginOrName: string;
         let loginStream: Observable<{ text: string; entities: User[] }>;
-        let searchStub: SinonStub;
+        let searchStub: jest.SpyInstance;
 
         beforeEach(() => {
             loginOrName = 'testLoginOrName';
             loginStream = of({ text: loginOrName, entities: [] });
-            searchStub = sinon.stub(userService, 'search');
+            searchStub = jest.spyOn(userService, 'search').mockImplementation();
+        });
+
+        afterEach(() => {
+            jest.restoreAllMocks();
         });
 
         it('should search users for given login or name', () => {
-            searchStub.returns(of(new HttpResponse({ body: [courseGroupUser] })));
+            searchStub.mockReturnValue(of(new HttpResponse({ body: [courseGroupUser] })));
             comp.searchAllUsers(loginStream).subscribe((users: any) => {
-                expect(users).to.deep.equal([courseGroupUser]);
+                expect(users).toEqual([courseGroupUser]);
             });
-            expect(searchStub).to.have.been.calledWith(loginOrName);
+            expect(searchStub).toHaveBeenCalledWith(loginOrName);
```

Comment:
Please also try to define the number of calls above.
Probably 1.

Reply:
Please search yourself for further occurrences.

Reply:
How could I specify the parameter and a number of calls at the same time? Or do you mean to check the number of calls in a separate `expect`?

Reply:
Separate, yes.

Reply:
I added them.